lib/encryption_algorithm.rb

```diff
@@ -54,33 +48,14 @@ def shift(code, shift)
     end
    end
 
-   def de_shift(code, shift)
-     code.map do |number|
-      if number.is_a?(Integer)
-        ( number - shift % 27) % 27
-      else
-         number
-      end
-    end
-   end
-
-   def zip_together(message, a_shift, b_shift, c_shift, d_shift)
-     one_code = shift(split_4th(message)[0], a_shift)
-     two_code = shift(split_4th(message)[1], b_shift)
-     three_code = shift(split_4th(message)[2], c_shift)
-     four_code = shift(split_4th(message)[3], d_shift)
+   def zip_together(message, a_shift, b_shift, c_shift, d_shift, multiplier = 1)
+     one_code = shift(split_4th(message)[0], a_shift, multiplier)
```

Comment:
1 method instead of 2. multiplier being an optional argument that can move cipher in reverse if -1 entered